swag/restswag.py

```diff
@@ -0,0 +1,141 @@
+        if getattr(controller[0], 'get_delete', None):
+            methodtuple = (controller[1] +  "/{" +
+                               getattr(controller[0], 'get_delete')._pecan['argspec'][0][1] + "}",
+                           ('delete', getattr(controller[0], 'get_delete')))
+            methods = add_method(methods, methodtuple)
+        if getattr(controller[0], 'delete', None):
+            methodtuple = (controller[1] +  "/{" +
+                               getattr(controller[0], 'delete')._pecan['argspec'][0][1] + "}",
+                           ('delete', getattr(controller[0], 'delete')))
+            methods = add_method(methods, methodtuple)
+        if getattr(controller[0], 'index', None):
+            pass
+        if getattr(controller[0], '_default', None):
+            pass
+        # pecan's _lookup function
+        if getattr(controller[0], '_lookup', None):
+            pass
+
+        # if controller uses custom routing
+        if getattr(controller[0], '_custom_actions', None):
+            for k in controller[0]._custom_actions:
+                m = getattr(controller[0], k, None)
+                if m:
+                    ml = [method in m]
+                    for method in ml:
+                        if method == 'get_one':
+                            methodtuple = (controller[1] + "/{" +
+                                           getattr(controller[0], 'get_one')._pecan['argspec'][0][1] + "}",
+                           ('get', getattr(controller[0], 'get_one')))
+                            methods = add_method(methods, methodtuple)
+                            pass
+                        elif method == 'get_delete':
+                            methodtuple = (controller[1] + "/{" +
+                                           getattr(controller[0], 'get_delete')._pecan['argspec'][0][1] + "}",
+                           ('delete', getattr(controller[0], 'get_delete')))
+                            methods = add_method(methods, methodtuple)
+                            pass
+                        elif method == 'delete':
+                            methodtuple = (controller[1] + "/{" +
+                                           getattr(controller[0], 'get_delete')._pecan['argspec'][0][1] + "}",
+                           ('delete', getattr(controller[0], 'delete')))
+                            methods = add_method(methods, methodtuple)
+                        else:
+                            methodtuple = (controller[1], (str(method).lower(), getattr(controller[0], k)))
+                            methods = add_method(methods, methodtuple)
+
+    return methods
+
+def add_method(methods, methodtuple):
+    path = methodtuple[0]
+    methodname = methodtuple[1][0]
+    methodfunction = methodtuple[1][1]
+    if path not in methods:
+        methods[path] = {}
+    methods[path][methodname] = methodfunction
+    return methods
+
+
+
+import ast
```

Comment:
we don't need it and the method at all, I used it while experimenting with _lookup